pkg/controller/connectionsecret/ensuresecret.go

```diff
@@ -6,7 +6,7 @@ import (
 	"net/url"
 
 	corev1 "k8s.io/api/core/v1"
-	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
```

Comment:
Question: What is the rational for changing this?

Reply:
It comes from a comment by @s-urbaniak  here:
https://github.com/mongodb/mongodb-atlas-kubernetes/pull/1798#discussion_r1741512235

I decided to make it a separate PR

Reply:
Ok, so we want to keep with the naming convention set by k8s. This new style is less readable but I agree consistency is better.

Can you create a ticket to associate this with?

Reply:
Done CLOUDP-271316

Reply:
BTW `apiErrors` is camel case and not considered idiomatic as a package alias name:
https://go.dev/blog/package-names

Reply:
Aha, noted, thanks for pointing me to that!

LGTM from me then